group/assignment6/move_cars.py

```diff
@@ -0,0 +1,38 @@
+def car_positions(state):
+    """ given state returns array where ith element is the position of ith car """
+    positions = [0] * len(start)
+    for place, car in enumerate(start):
+        positions[car] = place
+    return positions
+
+
+def generate_moves(start, final):
+    """ ith element of start state is the car currently in the ith space
+    ith element of final state is the car that should be in ith space
+    empty space is 0
+    yields optimal sequence of moves in format (car number, old space, new space) """
+
+    pos = car_positions(start)
+    for car, place in enumerate(pos):
+        if final[place] == car:
+            continue
+
+        if car:  # this is a cycle fix
+            yield car, place, pos[0]
+            pos[0], pos[car] = pos[car], pos[0]
+
+        car = final[pos[0]]
+        while car:
+            yield car, pos[car], pos[0]
```

Comment:
1) So, if I understand correctly, for now your function just swaps each car with car at its final position until the positions of all the cars will be coincide with final. But I don't understand, why we need the recursion for this? I am confused. I'm not sure if it will work properly for now (I didn't test it yet).
2) Yes, swaping car1 and car2 can be considered as some simplification of 3 moves: car1 -> empty space; car2 -> car1's prev. position; car1 -> car2's prev. position. But it seems like we should yield all these 3 moves separately for assignment purpose.